### PR TITLE
Epmrpp 33465 react.dashboard.widget.flaky test

### PR DESCRIPTION
**Pull request by gvenill**
Flaky tests and Most Failed tests widgets

### CODE REVIEW

File: app/src/common/css/variables/colors.scss [hunk 1]
```diff
@@ -30,6 +30,7 @@ $COLOR--light-yellow: #f5f9ec;
 $COLOR--orangish: #fb7848;
 $COLOR--tealish: #39c2d7;
 $COLOR--tealish-hover: #d7f3f7;
+$COLOR--tealish-light: #d7f2f7;
```
Comment (ArtsiomYeliseyenka) [inline app/src/common/css/variables/colors.scss]: Do not add any color variables without UX-Designer approve. The added color the same as $COLOR--tealish-hover, so there is no need to add new one.

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/testsTableBody.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { func } from 'prop-types';
+import classNames from 'classnames/bind';
+import { PTTests, PTColumns } from '../pTypes';
+import TestsTableRow from './testsTableRow';
+import matrixFactory from './matrix';
+import styles from '../testsTableWidget.scss';
+
+const cx = classNames.bind(styles);
+
+class TestsTableBody extends React.Component {
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/testsTableBody.jsx]: Please try to use existing common grid component.

File: app/src/components/main/absRelTime/absRelTime.jsx [hunk 3]
```diff
@@ -46,14 +30,17 @@ export class AbsRelTime extends Component {
...
-          {dateFormat(this.props.startTime)}
+          <FormattedRelative value={this.props.startTime} />
         </span>
+        <span className={cx('absolute-time')}>{dateFormat(this.props.startTime)}</span>
       </div>
     );
   }
 }
+
+export default AbsRelTime;
```
Comment (DiscoElevator) [inline app/src/components/main/absRelTime/absRelTime.jsx]: Please use named exports

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/index.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  </div>
+);
+
+TestsTable.propTypes = {
+  tests: PTTests.isRequired,
+  nameClickHandler: func.isRequired,
+  columns: PTColumns.isRequired,
+};
+
+export default TestsTable;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/index.jsx]: Please use named exports

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/count.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    {count} <FormattedMessage id="Common.of" defaultMessage="of" /> {total}
+  </div>
+);
+
+Count.propTypes = {
+  count: number.isRequired,
+  total: number.isRequired,
+};
+
+export default Count;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/count.jsx]: Please use named exports

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/launchInfoBlock.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+LaunchInfoBlock.propTypes = {
+  launchName: string.isRequired,
+  issueType: element,
+};
+
+LaunchInfoBlock.defaultProps = {
+  issueType: null,
+};
+
+export default LaunchInfoBlock;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/launchInfoBlock.jsx]: Please use named exports

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTableWidget.scss [hunk 7]
```diff
@@ -0,0 +1,211 @@
+.tests-table-widget {
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTableWidget.scss]: Please split styles by corresponding components

File: app/src/pages/inside/dashboardPage/widgets/flakyTests/flakyTests.jsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  />
+);
+
+FlakyTests.propTypes = {
+  tests: PTypes.PTTests.isRequired,
+  launch: PTypes.PTLaunch.isRequired,
+  nameClickHandler: func.isRequired,
+};
+
+export default FlakyTests;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/flakyTests/flakyTests.jsx]: Please use named exports

File: app/src/pages/inside/dashboardPage/widgets/mostFailedTests/mostFailedTests.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        launchName={launch.name}
+        nameClickHandler={nameClickHandler}
+        issueType={this.getIssueTypeMessage(issueType)}
+        columns={cfg.columns}
+      />
+    );
+  }
+}
+
+export default MostFailedTests;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/mostFailedTests/mostFailedTests.jsx]: Please use named exports

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/matrix.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+  Matrix.propTypes = {
+    tests: arrayOf(oneOfType([bool, PTStatus])).isRequired,
+    id: string.isRequired,
+  };
+
+  return Matrix;
+}
+
+export default matrixFactory;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTable/matrix.jsx]: Please use named exports

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  percentage: string.isRequired,
+  isFailed: arrayOf(bool),
+  total: number.isRequired,
+  failedCount: number,
+  switchCounter: number,
+  statuses: arrayOf(PTStatus),
+});
+
+export const PTIssueType = oneOf([
+  'statistics$executions$failed',
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: use constants from common/constants/statistics.js

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/launchInfoBlock/index.jsx [hunk 12]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { string, element } from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import classNames from 'classnames/bind';
+import styles from './launchInfoBlock.scss';
+
+const cx = classNames.bind(styles);
+
+const LaunchInfoBlock = ({ launchName, issueType }) => (
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/launchInfoBlock/index.jsx]: export const LaunchInfoBlock = ...

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js [hunk 13]
```diff
@@ -0,0 +1,52 @@
+import { string, node, shape, arrayOf, bool, number, oneOf } from 'prop-types';
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: Please follow the project code style and describe propTypes directly in component file

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+export const PTLaunch = shape({
+  id: string.isRequired,
+  name: string.isRequired,
+  number: string.isRequired,
+  issueType: string,
+});
+
+export const PTStatus = oneOf(['FAILED', 'PASSED', 'SKIPPED']);
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: Use constants from common/constants/launchStatuses.js and str.toUppercase()

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTableWidget.jsx [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+import { func, string, element } from 'prop-types';
+import classNames from 'classnames/bind';
+import { LaunchInfoBlock } from './launchInfoBlock';
+import { TestsTable } from './testsTable';
+import { PTTests, PTColumns } from './pTypes';
+import styles from './testsTableWidget.scss';
+
+const cx = classNames.bind(styles);
+
+function TestsTableWidget(props) {
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTableWidget.jsx]: Please follow project code style. export const TestsTableWidget = ({ ... }) => {

**Comment by ArtsiomYeliseyenka** [comment]
In a fact, only code style claims from my side.

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js (hunk 13 above)
Comment (gvenill) [reply app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: Removed propTypes that weren't used across multiple component

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js (hunk 11 above)
Comment (gvenill) [reply app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: Fixed

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/launchInfoBlock/index.jsx (hunk 12 above)
Comment (gvenill) [reply app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/launchInfoBlock/index.jsx]: Fixed

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js (hunk 14 above)
Comment (gvenill) [reply app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: Fixed

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTableWidget.jsx (hunk 15 above)
Comment (gvenill) [reply app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/testsTableWidget.jsx]: Fixed

File: app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js [hunk 16]
```diff
@@ -1,4 +1,5 @@
 import { string, node, shape, arrayOf, bool, number, oneOf } from 'prop-types';
+import { FAILED, PASSED, SKIPPED } from '../../../../../../common/constants/launchStatuses';
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: We have alias for 'common' folder. No need of ../../../../../../
Please fix all following cases.

Comment (gvenill) [reply app/src/pages/inside/dashboardPage/widgets/components/testsTableWidget/pTypes.js]: Fixed

**Pull request merged**